packages/block-library/src/navigation/index.php

```diff
@@ -253,8 +253,15 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 
 	$inner_blocks = $block->inner_blocks;
 
-	// If `__unstableLocation` is defined, create inner blocks from the classic menu assigned to that location.
-	if ( empty( $inner_blocks ) && array_key_exists( '__unstableLocation', $attributes ) ) {
+	// If `__unstableLocation` is defined and:
+	// - we don't have a relationship to a `wp_navigation` Post (via `navigationMenuId`).
+	// - we don't have any inner blocks (e.g. a block pattern).
+	// ...then create inner blocks from the classic menu assigned to that location.
+	if (
+		array_key_exists( '__unstableLocation', $attributes ) &&
+		! array_key_exists( 'navigationMenuId', $attributes ) &&
+		empty( $inner_blocks )
+	) {
 		$menu_items = gutenberg_get_menu_items_at_location( $attributes['__unstableLocation'] );
 		if ( empty( $menu_items ) ) {
 			return '';
```

Comment:
```suggestion
			return $inner_blocks;
```

Reply:
I don't think that will work.

Reply:
Yeah, this returns the actual rendered content. It will get a bit tricky, probably need to use an if/else.

Reply:
Ah yes good point.